modules/security-jwt/src/main/java/org/opencastproject/security/jwt/DynamicLoginHandler.java

```diff
@@ -261,11 +268,78 @@ private String extractEmail(DecodedJWT jwt) {
   private Set<JpaRole> extractRoles(DecodedJWT jwt) {
     JpaOrganization organization = fromOrganization(securityService.getOrganization());
     Set<JpaRole> roles = new HashSet<>();
-    for (String mapping : roleMappings) {
-      String role = evaluateMapping(jwt, mapping, false);
+    Consumer<String> addRole = (String role) -> {
       if (StringUtils.isNotBlank(role)) {
         roles.add(new JpaRole(role, organization));
       }
+    };
+
+    // Evaluate the standard role mapping if specified
+    if (ocStandardRoleMappings) {
+      // Read `role` claim
+      try {
+        var rolesClaim = jwt.getClaim("roles");
+        if (rolesClaim != null && !rolesClaim.isNull()) {
+          for (String r : rolesClaim.asArray(String.class)) {
+            if (!r.isBlank()) {
+              addRole.accept("ROLE_" + r);
+            }
+          }
+        }
+      } catch (JWTDecodeException e) {
+        logger.warn("claim 'roles' is not an array of strings, ignoring");
```

Comment:
I would suggest using `logger.debug` (also below). This shouldn't happen if the integration is using a correct schema, but if not this will fill the logs for adopters.